ci-operator/step-registry/rosa/conf/machine-pool/rosa-conf-machine-pool-commands.sh

```diff
@@ -0,0 +1,148 @@
+  if [ $? -ne 0 ]; then
+    echo "Login failed"
+    exit 1
+  fi
+else
+  echo "Cannot login! You need to specify the offline token ROSA_TOKEN!"
+  exit 1
+fi
+
+# Switches
+LABELS="prowci=true"
+if [[ ! -z "$MP_LABELS" ]]; then
+  LABELS="${LABELS},${MP_LABELS}"
+fi
+
+TAINTS="prowci=true:NoSchedule"
+if [[ ! -z "$MP_TAINTS" ]]; then
+  TAINTS="${TAINTS},${MP_TAINTS}"
+fi
+
+MP_NODES_SWITCH=""
+if [[ "$MP_ENABLE_AUTOSCALING" == "true" ]]; then
+  MP_NODES_SWITCH="--enable-autoscaling --min-replicas ${MP_MIN_REPLICAS} --max-replicas ${MP_MAX_REPLICAS}"
+else
+  MP_NODES_SWITCH="--replicas ${MP_REPLICAS}"
+fi
+
+TUNING_CONFIG_SWITCH=""
+if [[ "$USE_TUNING_CONFIG" == "true" ]]; then
+  tuning_config_file=$(cat "${SHARED_DIR}/tuning_config_file")
+  TUNING_CONFIG_SWITCH="--tuning-configs ${tuning_config_file}"
+fi
+
+SPOT_INSTANCES_SWITCH=""
+if [[ "$USE_SPOT_INSTANCES" == "true" ]]; then
+  SPOT_INSTANCES_SWITCH="--use-spot-instances --spot-max-price ${SPOT_MAX_PRICE}"
+fi
+
+LOCAL_ZONE_SWITCH=""
+if [[ "$LOCAL_ZONE" == "true" ]]; then
+  LOCAL_ZONE_SWITCH=""
+  private_subnet_id=$(cat ${SHARED_DIR}/private_subnet_ids | tr -d "[']" | cut -d ',' -f 1)
+  if [[ -z "${private_subnet_id}" ]]; then
+    echo -e "The private_subnet_id is mandatory."
+    exit 1
+  fi
+
+  LOCAL_ZONE_SWITCH="--subnet ${private_subnet_id}"
+fi
+
+# Create machine pool on the cluster
+function createMachinepool() {
+  subfix=$(openssl rand -hex 2)
+  MP_NAME="mp-$subfix"
+  
+  ZONE=$1
+  AZ_SWITCH=""
+  if [[ ! -z "$ZONE" ]]; then
+    AZ_SWITCH="--availability-zone ${ZONE}"
+  fi
+
+  AUTO_REPAIR_SWITCH=""
+  if [[ "$HOSTED_CP" == "true" ]]; then
+    AUTO_REPAIR_SWITCH="--autorepair"
+  fi
+
+  echo -e " Create machine pool ${MP_NAME} on the cluster ${CLUSTER_ID} ...
+rosa create machinepool -y \
+-c ${CLUSTER_ID}
+--name ${MP_NAME} \
+--instance-type ${MP_MACHINE_TYPE} \
+--labels ${LABELS} \
+--taints ${TAINTS} \
+${MP_NODES_SWITCH} \
+${TUNING_CONFIG_SWITCH} \
+${SPOT_INSTANCES_SWITCH} \
+${AZ_SWITCH} \
+${LOCAL_ZONE_SWITCH} \
+${AUTO_REPAIR_SWITCH}
+"
+
+rosa create machinepool -y \
+                        -c ${CLUSTER_ID} \
+                        --name ${MP_NAME} \
+                        --instance-type ${MP_MACHINE_TYPE} \
+                        --labels ${LABELS} \
+                        --taints ${TAINTS} \
+                        ${MP_NODES_SWITCH} \
+                        ${TUNING_CONFIG_SWITCH} \
+                        ${SPOT_INSTANCES_SWITCH} \
+                        ${AZ_SWITCH} \
+                        ${LOCAL_ZONE_SWITCH} \
+                        ${AUTO_REPAIR_SWITCH}
+} 
+
+
+if [[ ! -z "$MP_ZONE" ]]; then
+  MP_ZONE="${CLOUD_PROVIDER_REGION}${MP_ZONE}"
+fi
+
+if [[ "$HOSTED_CP" == "false" ]]; then
+  createMachinepool "$MP_ZONE"
+else
+  ZONES=$(rosa describe cluster -c ${CLUSTER_ID} -o json | jq -r ".nodes.availability_zones[]")
+  for i in ${ZONES}; do
+    createMachinepool "$i"
+  done
+fi
```

Comment:
Is it possible to use same logic to create machinepool in all zones of the nodes on both hosted and non-hosted clusters?

Reply:
For the non-sts, if you do not specify `LOCAL_ZONE` or `MP_ZONE`, the machine pool will be created with the cluster attribute, etc. The multiAZ MP will be created on the multiAZ rosa cluster, and the singleAZ MP will be created on the singleAZ rosa cluster.